app/services/find/course_type_answer_determiner.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module Find
+  class CourseTypeAnswerDeterminer
+    attr_reader :age_group, :visa_status, :university_degree_status
+
+    def initialize(age_group:, visa_status:, university_degree_status:)
+      @age_group = age_group
+      @visa_status = visa_status
+      @university_degree_status = university_degree_status
+    end
+
+    def show_undergraduate_courses?
+      !further_education? && university_degree_status.present? && no_degrees? && does_not_require_visa_sponsorship?
+    end
+
+    def not_elibible_for_undergraduate_courses?
+      !further_education? && university_degree_status.present? && no_degrees? && require_visa_sponsorship?
+    end
+
+    private
+
+    def further_education?
+      age_group == 'further_education'
+    end
+
+    def require_visa_sponsorship?
+      ActiveModel::Type::Boolean.new.cast(visa_status)
+    end
+
+    def does_not_require_visa_sponsorship?
+      !require_visa_sponsorship?
+    end
+
+    def no_degrees?
+      !degrees?
+    end
+
+    def degrees?
+      ActiveModel::Type::Boolean.new.cast(university_degree_status)
```

Comment:
```suggestion
      university_degree_status.present? && ActiveModel::Type::Boolean.new.cast(university_degree_status)
```

I think it might make the public methods slightly easier to read this way and the presence check is required for for this method to work at all, so it makes sense to put it down here.

Reply:
It won't work because for this specific cycle, the degree question won't even exist so I have to maintain the check exactly there. 

I could move this check to `universisty_degree_question_answered?`